src/js/archive.js

```diff
@@ -10,47 +10,36 @@ function onArchiveLoad() {
 
 // ARCHIVE PAGE FUNCTIONS
 
+
+/* eslint-disable no-undef */
 function populateArchive() {
   loadPlatformsThenData(function() {
 
     // TODO - the commented-out repoName variable below should be passed into loadJSON below as the first argument, replacing openjdk-releases.
     // This can only be done after the repository name is updated from 'openjdk-releases' to 'openjdk8-releases'.
-
-    var repoName = (variant + '-releases');
-
-    loadJSON(repoName, 'releases', function(response) {
-      function checkIfProduction(x) { // used by the array filter method below.
-        return x.prerelease === false && x.assets[0];
-      }
-
-      // Step 1: create a JSON from the XmlHttpRequest response
-      // Step 2: filter out all releases from this JSON that are marked as 'pre-release' in GitHub.
-      var releasesJson = JSON.parse(response).filter(checkIfProduction);
-
+    var handleResponse = function(response, oldRepo) {
       // if there are releases prior to the 'latest' one (i.e. archived releases)...
-      if (typeof releasesJson[0] !== 'undefined') {
-        loadJSON(repoName, 'jck', function(response_jck) {
-          var jckJSON = {}
-          if (response_jck !== null){
-            jckJSON = JSON.parse(response_jck)
-          }
-          buildArchiveHTML(releasesJson, jckJSON);
-        });
-      } else { // if there are no releases (beyond the latest one)...
-        // report an error, remove the loading dots
-        loading.innerHTML = '';
-        errorContainer.innerHTML = '<p>There are no archived releases yet! See the <a href=\'./releases.html?variant=' + variant + '\'>Latest release</a> page.</p>';
+      if (response !== null) {
+        buildArchiveHTML(response);
+      } else {
+        if(oldRepo) {
```

Comment:
Not sure why this logic is wrapped in oldRepo?

Reply:
The sequence is we check the new repo, if there are no files then check the old one, this then says if we find nothing from the old, display an error